src/status_im/contexts/wallet/events.cljs

```diff
@@ -546,3 +550,43 @@
  :wallet/process-watch-only-account-from-backup
  (fn [_ [{:keys [backedUpWatchOnlyAccount]}]]
    {:fx [[:dispatch [:wallet/process-account-from-signal backedUpWatchOnlyAccount]]]}))
+
+(defn process-keypairs
+  [{:keys [db]} [keypairs]]
+  (let [existing-keypairs-by-id               (get-in db [:wallet :keypairs])
+        existing-accounts-by-address          (get-in db [:wallet :accounts])
+        {:keys [removed-keypair-ids
+                removed-account-addresses
+                updated-keypairs-by-id
+                updated-accounts-by-address]} (data-store/process-keypairs keypairs)
+        updated-keypair-ids                   (set (keys updated-keypairs-by-id))
+        updated-account-addresses             (set (keys updated-accounts-by-address))
+        existing-account-addresses            (set (keys existing-accounts-by-address))
+        new-account-addresses                 (clojure.set/difference updated-account-addresses
+                                                                      existing-account-addresses)
+        old-account-addresses                 (->> (vals existing-accounts-by-address)
+                                                   (filter (fn [{:keys [address key-uid]}]
+                                                             (and (contains? updated-keypair-ids key-uid)
+                                                                  (not (contains?
+                                                                        updated-accounts-by-address
+                                                                        address)))))
+                                                   (map :address))]
+    (cond-> {:db (-> db
+                     (assoc-in [:wallet :keypairs]
+                               (-> (apply dissoc existing-keypairs-by-id removed-keypair-ids)
+                                   (merge updated-keypairs-by-id)))
+                     (assoc-in [:wallet :accounts]
+                               (-> (apply dissoc
+                                          existing-accounts-by-address
+                                          (into removed-account-addresses old-account-addresses))
+                                   (merge updated-accounts-by-address))))}
+      (seq new-account-addresses)
+      (assoc :fx
+             (->> new-account-addresses
+                  (mapcat (fn [address]
+                            [[:dispatch [:wallet/get-wallet-token-for-account address]]
+                             [:dispatch
+                              [:wallet/request-new-collectibles-for-account-from-signal address]]
+                             [:dispatch [:wallet/check-recent-history-for-account address]]])))))))
```

Comment:
We can skip the thread-last and use the `new-account-addresses` directly in `mapcat` here.